src/azure-cli/azure/cli/command_modules/resource/_params.py

```diff
@@ -137,6 +137,7 @@ def load_arguments(self, _):
         c.argument('resource_provider_namespace', options_list=['--namespace', '-n'], completer=get_providers_completion_list, help=_PROVIDER_HELP_TEXT)
 
     with self.argument_context('provider register') as c:
+        c.argument('mg', arg_type=management_group_id_type, help="The management group id to register.", required=False, options_list=['--mg'])
```

Comment:
`, required=False, options_list=['--mg']` these two parameters don't seem necessary. Could you please try to delete them?

Reply:
I added them so they show up in the --help arguments section for provider register. If I remove them instead of 
`--mg                      : The management group id to register.`
it will read:
`--management-group-id -m [Required] : The management group id to register.`

which isn't what I want since it's the wrong flag and not required

Reply:
I can remove the `arg_type` as well and manually add the options list if that would be a better solution

Reply:
In fact, `management_group_id_type` doesn't fit your scenario, so please don't use it
Your scenario is:
```
c.argument('mg', help="The management group id to register.")
```
But `management_group_id_type` is defined as
```
management_group_id_type = CLIArgumentType(options_list=['--management-group-id', '-m'], required=True, help='The management group id.')
```

Reply:
By the way, do we need to consider keeping parameter names consistent? How about naming the parameter as `['--management-group-id', '-m']` to users?

Reply:
Yeah that makes more sense I'll make that switch